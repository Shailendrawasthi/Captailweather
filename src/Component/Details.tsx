
import { Container, Typography, Button } from '@mui/material';
import { Box } from '@mui/system';
import axios from 'axios';
import React, { useState } from 'react';
import { useLocation, Link } from 'react-router-dom';



const Details = () => {
    let API_KEY = '80b3a297fb887d9058ba106509dd2d03';
    const { state }: any = useLocation();
    const [country, setCountry] = useState<any>("");
    const [weatherData, setWeatherData] = useState<any>("");
    const data = state?.data;
    console.log(data);
    const weatherDetail = async (capital: any) => {
        setCountry(capital);
        const resp = await axios.get
            (`http://api.weatherstack.com/current?access_key=${API_KEY}&query=${capital}`);
        setWeatherData(resp.data);
        console.log(resp.data);


    }
    function renderWeatherDetails() {
        return (
            <div>
                <Typography>Weather Information</Typography>
                <div>
                    <div>

                        <img
                            src={weatherData?.current?.weather_icons[0]}
                            alt=""
                        />
                    </div>
                    <div className="weather-info">
                        Temperature
                        <div className="weather-data">
                            {weatherData?.current?.temperature}
                        </div>
                    </div>
                    <div className="weather-info">
                        Wind Speed
                        <div className="weather-data">
                            {weatherData?.current?.wind_speed}
                        </div>
                    </div>
                    <div className="weather-info">
                        Precip
                        <div className="weather-data">
                            {weatherData?.current?.precip}
                        </div>
                    </div>
                </div>
                <div>
                    <Link style={{ textDecoration: 'none' }}
                        id='go-back' to="/">
                        <Button>Back</Button>
                    </Link>
                </div>
            </div>
        )
    }
    return (
        <div>
            <Container>
                <Box>
                    <Typography variant='h4'>Country Details</Typography>
                    {data?.map((item: any) => {
                        return (


                            <div>
                                <img
                                    src={item?.flags.png}
                                    alt={item?.flag}
                                    height="140"
                                    width="200"
                                />
                                <div>
                                    <strong>Capital :-</strong>
                                    {item?.capital}
                                </div>
                                <div>
                                    <strong>Population :-</strong>
                                    {item?.population}
                                </div>
                                <div>
                                    <strong>   Latlng :-</strong>
                                    {item?.capitalInfo.latlng[0]}, {item?.capitalInfo.latlng[1]}
                                </div>
                                <Button id='Weather' onClick={() => weatherDetail(item?.capital)}>Capital Weather</Button>
                                {String(weatherData?.location?.name) === String(item?.capital) && weatherData ? renderWeatherDetails() : null}

                            </div>


                        );
                    })}
                </Box>
            </Container>
        </div>)
};
export default Details;

