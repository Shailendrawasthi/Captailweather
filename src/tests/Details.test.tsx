import { shallow } from "enzyme";
import Details from "../Component/Details";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useLocation: () => ({
      location: jest.fn().mockImplementation(() => ({})),
    }),
}));

describe('testing Details component', () => {

    let wrapper: any;

    beforeEach(() => {
        wrapper = shallow(<Details />);
    });

    it('renders  component without crashes', () => {
        expect(wrapper.exists()).toBe(true);
    });

    it('should render a <div />', () => {
        const result = wrapper.find('div').length;
        expect(result).toEqual(1);
    });
    it('should render a <div />', () => {
        const result = wrapper.find('Typography').length;
        expect(result).toEqual(0);
    });

    it('checking Submit button', () => {
        expect(wrapper.text('Capital Weather')).toBeTruthy();
    });
    it('renders snapshot for DetailsInfo component', () => {
        expect(wrapper).toMatchSnapshot();
    });
})

