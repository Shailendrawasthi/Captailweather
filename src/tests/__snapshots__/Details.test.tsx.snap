// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Details component renders snapshot for CountryInfo component 1`] = `ShallowWrapper {}`;

exports[`testing Details component renders snapshot for DetailsInfo component 1`] = `ShallowWrapper {}`;
